columnflow/calibration/cms/jets.py

```diff
@@ -617,23 +620,23 @@ def jer(self: Calibrator, events: ak.Array, **kwargs) -> ak.Array:
         events.Rho.fixedGridRhoFastjetAll
     )
 
+    # variable naming convention
+    variable_map = {
+        "JetEta": events.Jet.eta,
+        "JetPt": events.Jet.pt,
+        "Rho": rho,
+    }
+
     # pt resolution
-    jer = ak_evaluate(
-        self.evaluators["jer"],
-        events.Jet.eta,
-        events.Jet.pt,
-        rho,
-    )
+    inputs = [variable_map[inp.name] for inp in self.evaluators["jer"].inputs]
+    jer = ak_evaluate(self.evaluators["jer"], *inputs)
 
     # JER scale factors and systematic variations
-    jersf = {
-        syst: ak_evaluate(
-            self.evaluators["sf"],
-            events.Jet.eta,
-            syst,
-        )
-        for syst in ("nom", "up", "down")
-    }
+    jersf = {}
+    for syst in ("nom", "up", "down"):
+        variable_map["systematic"] = syst
```

Comment:
This does the job but it's a bit messy IMO to update the `variable_map` from outside to loop in this way. Just in case in some future version of the code we decide to rely on the `variable_map` further down, it might lead to errors since the map ends up with the `systematic` key set to "down". It might be better to create a local copy adding the `systematic` key, e.g. something like this:
```suggestion
        variable_map = dict(variable_map, systematic=syst)
```

Reply:
Good point, added with the latest commit